web/Services/PropertyService.cs

```diff
@@ -219,9 +219,15 @@ public string CreateNewPublicViewLink(int propertyId)
     
     public async Task<List<PropertyDbModel>?> SortPropertiesByLocation(string postalCode, int page, int perPage)
     {
-        var responseBody = await _postcodeApiService.MakeApiRequestToPostcodeApi(postalCode);
-        var model = _postcodeApiService.TurnResponseBodyToModel(responseBody);
-        if (model.Result == null || model.Result.Lat == null || model.Result.Lon == null)
+        var postcode = _postcodeService.FormatPostcode(postalCode);
+        var postcodeList = new List<string> { postcode };
+        await _postcodeService.AddPostcodesToDatabaseIfAbsent(postcodeList);
+        if (postcode == "")
+        {
+            return null;
+        }
+        var model = _dbContext.Postcodes.SingleOrDefault(p => p.Postcode == postcode);

```

Comment:
I have jsut realised, can we not combine these 2? as in if the postcode is empty string the model will be null

Reply:
I thought that to, but it was causing an 'Object reference not set to an instance of an object error' so I kept both checks in